src/pages/groupComparison/CreateGroupFromOverlap.tsx

```diff
@@ -0,0 +1,189 @@
+import * as React from "react";
+import {observer} from "mobx-react";
+import GroupComparisonStore from "./GroupComparisonStore";
+import DefaultTooltip from "../../shared/components/defaultTooltip/DefaultTooltip";
+import autobind from "autobind-decorator";
+import {action, computed, observable} from "mobx";
+import {SyntheticEvent} from "react";
+import {remoteData} from "../../shared/api/remoteData";
+import {
+    ComparisonGroup,
+    DUPLICATE_GROUP_NAME_MSG,
+    excludeSamples,
+    intersectSamples,
+    unionSamples
+} from "./GroupComparisonUtils";
+import {StudyViewFilter} from "../../shared/api/generated/CBioPortalAPIInternal";
+import {MakeMobxView} from "../../shared/components/MobxView";
+import LoadingIndicator from "../../shared/components/loadingIndicator/LoadingIndicator";
+import {SessionGroupData} from "../../shared/api/ComparisonGroupClient";
+import _ from "lodash";
+import {getStudiesAttrForPatientOverlapGroup, getStudiesAttrForSampleOverlapGroup, joinNames} from "./OverlapUtils";
+
+export interface ICreateGroupFromOverlapProps {
+    store:GroupComparisonStore;
+    includedRegions:string[][]; // group.uid[][]
+    allGroupsInVenn:string[]; // uid[]
+    x:number;
+    y:number;
+    submitGroup:(group:SessionGroupData)=>void;
+    caseType:"sample"|"patient";
+    width:number;
+}
+
+function getRegionSummary(
+    region:string[],
+    allGroupsInVenn:string[],
+    uidToGroup:{[uid:string]:ComparisonGroup},
+    caseType:"sample"|"patient"
+) {
+    const includedNames = region.map(uid=>uidToGroup[uid].name);
+    const excludedNames = _.difference(allGroupsInVenn, region).map(uid=>uidToGroup[uid].name);
+
+    let ret = <span>{caseType[0].toUpperCase()}{caseType.substring(1)}s in {joinNames(includedNames, "and")}</span>;
```

Comment:
could you use template strings in to avoid nesting <spans> ?

Reply:
joinNames bolds certain elements so it has to be JSX